server/rpc.go

```diff
@@ -89,12 +91,27 @@ func (s *RPC) Next(c context.Context, filter rpc.Filter) (*rpc.Pipeline, error)
 		}
 	}
 
+	var st *expr.Selector
+	var err error
+	if filter.Expr != "" {
```

Comment:
Thoughts on moving this logic into the `fn` declaration? That would reduce the need for having the extra variable declarations. Or is catching a potential error from the selector initialization important?

Reply:
yes, for caching, otherwise it would re-parse the sql string every time :)